### PR TITLE
ncnews 8

### PR DESCRIPTION
**Pull request by timwtuck**

### CODE REVIEW

**Review by giteron — CHANGES_REQUESTED** [review]
An inventive way to fulfil the needs of this ticket! However, having consulted with the other tutors, we agree - there are great efficiency gains to be made for your server by using a single DB query here, rather than one query per ID. I'll DM you :)

File: __tests__/app.test.js [hunk 1]
```diff
@@ -45,6 +45,33 @@ describe('# GET REQUESTS', () => {
...
+                    body.articles.forEach(article => {
+                        expect(article).toMatchObject({
+                            author: expect.any(String),
+                            title: expect.any(String),
+                            article_id: expect.any(Number),
+                            topic: expect.any(String),
+                            created_at: expect.any(String),
+                            votes: expect.any(Number),
+                            comment_count: expect.any(Number)
+                        });
```
Comment (jadelandeg) [inline __tests__/app.test.js:66]: consider checking that the array is the correct length

File: models/articles.models.js [hunk 2]
```diff
@@ -17,22 +18,31 @@ exports.selectArticle = async (id) => {
     return article;
 }
 
+exports.selectAllArticles = async () => {
+    
+    const query = `SELECT articles.*, COUNT(comment_id)::INTEGER AS comment_count FROM articles
+                    INNER JOIN comments ON comments.article_id = articles.article_id
```
Comment (jadelandeg) [inline models/articles.models.js:24]: is inner join getting back all of the articles or only the articles that have comments?

**Review by jadelandeg — COMMENTED** [review]
Just a couple of things to check - is this the right joint to use?

**Pull request merged**